### ISSUE TITLE
libreface 0.1.0 depends on mediapipe==0.9.3

### ISSUE THREAD

**Issue by Victor716** [ihp-lab/LibreFace#10]
when running pip install -r requirements_new.txt --upgrade libreface
find 

ERROR: Could not find a version that satisfies the requirement mediapipe==0.9.3 (from versions: 0.10.0, 0.10.1, 0.10.2, 0.10.3, 0.10.5, 0.10.7, 0.10.8, 0.10.9, 0.10.10, 0.10.11)

Given this I installed mediapipe 0.10.0

but it came with another err: libreface 0.1.0 depends on mediapipe==0.9.3

how to fix that?

**Comment by msoley** [comment]
@ac-alpha please take a look

**Comment by ac-alpha** [comment]
@Victor716 Actually, you can simply use `pip install --upgrade libreface` to install libreface. Which Python version are you using?

**Comment by wulispa** [comment]
> @Victor716 Actually, you can simply use to install libreface. Which Python version are you using?`pip install --upgrade libreface`

I also encounter this situation.  i try python3.7,3.8and3.9. when i use py3.9 with mediapipe==0.10.0. numpy and protobuf will be confilcted.

**Comment by wzh125** [comment]
the same problem.

**Comment by ET2780** [comment]
Same problem

**Comment by ac-alpha** [comment]
@ET2780 @wzh125 Can you please comment on your Python versions? Also, please post the output of the `pip freeze` command inside your Python environment so that we can reproduce the issue.

**Comment by ET2780** [comment]
python version: Python 3.11.3
my pip freeze:
absl-py==2.1.0
aiofiles==23.2.1
aiohttp==3.8.4
aiosignal==1.3.1
alembic==1.13.2
annotated-types==0.6.0
antlr4-python3-runtime==4.9.3
anyio==4.3.0
appnope==0.1.4
asttokens==2.4.1
astunparse==1.6.3
async-timeout==4.0.2
attrs==23.1.0
audiocraft==1.3.0
audioread==3.0.1
Authlib==1.3.1
av==11.0.0
azure-cognitiveservices-speech==1.34.0
babel==2.16.0
backcall==0.2.0
beautifulsoup4==4.12.3
bibtexparser==2.0.0b7
bidict==0.22.1
bleach==6.1.0
blinker==1.9.0
blis==0.7.11
build==1.2.1
cachelib==0.10.2
cachetools==5.3.3
catalogue==2.0.10
certifi==2024.12.14
cffi==1.16.0
charset-normalizer==3.4.1
click==8.1.7
clldutils==3.22.2
cloudpathlib==0.19.0
cloudpickle==3.0.0
colorama==0.4.6
coloredlogs==15.0.1
colorlog==6.8.2
comm==0.2.2
confection==0.1.5
contourpy==1.3.0
cryptography==42.0.8
csvw==3.3.1
ctranslate2==4.3.1
cycler==0.12.1
cymem==2.0.8
cytoolz==0.12.3
debugpy==1.8.5
decorator==5.1.1
defusedxml==0.7.1
demucs==4.0.1
distro==1.9.0
dlinfo==1.2.1
docopt==0.6.2
dora_search==0.1.12
duckduckgo_search==6.2.12
einops==0.8.0
encodec==0.1.1
executing==2.1.0
fastapi==0.114.2
faster-whisper==1.0.3
fastjsonschema==2.20.0
ffmpy==0.4.0
filelock==3.13.1
flashy==0.0.2
Flask==3.1.0
flask-auth0==0.1.0
Flask-Cors==4.0.0
Flask-Migrate==4.0.7
Flask-Session==0.5.0
Flask-SocketIO==5.3.4
Flask-SQLAlchemy==3.0.3
flatbuffers==24.12.23
fonttools==4.53.1
frozenlist==1.3.3
fsspec==2024.2.0
gast==0.6.0
gdown==5.2.0
golang==0.0.1
google-api-core==2.17.1
google-auth==2.28.1
google-cloud-speech==2.25.0
google-cloud-texttospeech==2.16.2
google-pasta==0.2.0
googleapis-common-protos==1.62.0
gradio==4.44.0
gradio_client==1.3.0
greenlet==3.0.3
grpcio==1.69.0
grpcio-status==1.62.0
gTTS==2.5.1
gunicorn==20.1.0
h11==0.14.0
h5py==3.12.1
httpcore==1.0.5
httpx==0.27.0
huggingface-hub==0.24.6
humanfriendly==10.0
hydra-colorlog==1.2.0
hydra-core==1.3.2
idna==3.10
importlib_resources==6.4.5
intervaltree==3.1.0
ipykernel==6.29.5
ipython==8.12.3
isodate==0.6.1
itsdangerous==2.2.0
jax==0.2.10
jaxlib==0.4.38
jedi==0.19.1
Jinja2==3.1.2
jiter==0.8.2
joblib==1.3.1
jsonschema==4.23.0
jsonschema-specifications==2023.12.1
julius==0.2.7
jupyter_client==8.6.2
jupyter_core==5.7.2
jupyterlab_pygments==0.3.0
keras==3.7.0
kiwisolver==1.4.7
lameenc==1.7.0
langcodes==3.4.0
language-tags==1.2.0
language_data==1.2.0
lazy_loader==0.4
lhotse @ git+https://github.com/lhotse-speech/lhotse@4ca97dc3e5da0a8ad96f3811c387b62b90d62422
libclang==18.1.1
librosa==0.10.2.post1
lightning-utilities==0.11.7
lilcom==1.8.0
llvmlite==0.43.0
lxml==5.3.0
Mako==1.3.5
marisa-trie==1.2.0
Markdown==3.7
markdown-it-py==3.0.0
MarkupSafe==3.0.2
matplotlib==3.9.2
matplotlib-inline==0.1.7
mdurl==0.1.2
mediapipe==0.10.20
mistune==3.0.2
ml-dtypes==0.3.1
more-itertools==10.2.0
mpmath==1.3.0
msgpack==1.1.0
multidict==6.0.4
murmurhash==1.0.10
namex==0.0.8
nbclient==0.10.0
nbconvert==7.16.4
nbformat==5.10.4
nest-asyncio==1.6.0
networkx==3.2.1
nltk==3.8.1
num2words==0.5.13
numba==0.60.0
numpy==1.26.4
omegaconf==2.3.0
onnxruntime==1.19.2
openai==1.59.3
openai-whisper==20231117
opencv-contrib-python==4.10.0.84
opencv-python==4.10.0.84
openunmix==1.3.0
opt_einsum==3.4.0
optree==0.13.1
orjson==3.10.7
packaging==24.2
pandas==2.2.2
pandocfilters==1.5.1
parso==0.8.4
pexpect==4.9.0
phonemizer==3.3.0
pickleshare==0.7.5
pillow==10.4.0
pip-tools==7.4.1
pipreqs==0.5.0
platformdirs==4.3.2
pooch==1.8.2
preshed==3.0.9
primp==0.6.2
prompt_toolkit==3.0.47
proto-plus==1.23.0
protobuf==4.25.5
psutil==6.0.0
ptyprocess==0.7.0
pure_eval==0.2.3
pyasn1==0.5.1
pyasn1-modules==0.3.0
pycparser==2.22
pydantic==2.7.1
pydantic_core==2.18.2
pydub==0.25.1
Pygments==2.19.0
pylatexenc==2.10
pyngrok==7.2.0
pyparsing==3.1.4
pyphen==0.14.0
pyproject_hooks==1.1.0
PySocks==1.7.1
python-dateutil==2.9.0.post0
python-dotenv==1.0.0
python-engineio==4.5.1
python-multipart==0.0.9
python-socketio==5.8.0
python-version==0.0.2
pytz==2024.2
PyYAML==6.0.2
pyzmq==26.2.0
rdflib==7.0.0
redis==5.0.0
referencing==0.35.1
regex==2023.6.3
requests==2.32.3
retrying==1.3.4
rfc3986==1.5.0
rich==13.9.4
rpds-py==0.20.0
rsa==4.9
ruff==0.6.5
safetensors==0.4.5
scikit-learn==1.5.2
scipy==1.14.1
segments==2.2.1
semantic-version==2.10.0
sentencepiece==0.2.0
shellingham==1.5.4
six==1.17.0
smart-open==7.0.4
sniffio==1.3.1
sortedcontainers==2.4.0
sounddevice==0.5.1
soundfile==0.12.1
soupsieve==2.6
soxr==0.5.0.post1
spacy==3.7.6
spacy-legacy==3.0.12
spacy-loggers==1.0.5
SQLAlchemy==2.0.31
srsly==2.4.8
stack-data==0.6.3
starlette==0.38.5
submitit==1.5.1
sympy==1.12
tabulate==0.9.0
tensorboard==2.16.2
tensorboard-data-server==0.7.2
tensorflow==2.16.2
tensorflow-addons==0.23.0
tensorflow-io-gcs-filesystem==0.37.1
termcolor==2.5.0
textblob==0.17.1
TextGrid==1.6.1
textstat==0.7.3
thinc==8.2.5
threadpoolctl==3.5.0
tiktoken==0.6.0
tinycss2==1.3.0
tokenizers==0.20.0
tomlkit==0.12.0
toolz==0.12.1
torch==2.1.0
torchaudio==2.1.0
torchdata==0.7.0
torchmetrics==1.4.2
torchtext==0.16.0
torchvision==0.16.0
tornado==6.4.1
tqdm==4.65.0
traitlets==5.14.3
transformers==4.45.0
treetable==0.2.5
typeguard==2.13.3
typer==0.12.5
typing_extensions==4.12.2
tzdata==2024.1
uritemplate==4.1.1
urllib3==2.3.0
uvicorn==0.30.6
wasabi==1.1.3
wcwidth==0.2.13
weasel==0.4.1
webencodings==0.5.1
websockets==12.0
Werkzeug==3.1.3
whisper==1.1.10
wrapt==1.17.0
xformers==0.0.22.post7
yarg==0.1.9
yarl==1.9.1

**Comment by ac-alpha** [comment]
@ET2780 Can you please try installing on python==3.8? We have not tested LibreFace on other Python versions yet.

**Comment by 6namdang** [comment]
had the same issue when downloading libreface. Running it on python3.8

**Comment by jerzytrcka** [comment]
I had the same issue on python 3.8. 
As far as I can tell I fixed it by changing the mediapipe version to 0.10.5. 
I forked the repo and changed the requirements_new.txt entry, then the install was successful

**Comment by XaingHui** [comment]
the pip has not mediapipe == 0.9.3.
so you should compile source code mediapipe which version is 0.9.3

**Comment by joos2010kj** [comment]
For now it works with mediapipe==0.10.5

Do as follows:

`conda create -n libreface python=3.8`
`conda activate libreface`
`git clone https://github.com/ihp-lab/LibreFace.git`
`cd LibreFace`
`vi requirements_new.txt`
Type in `:%s/0.9.3/0.10.5/g` then `:wq`
`conda install ffmpeg`
`pip install .`

**Comment by 6namdang** [comment]
> For now it works with mediapipe==0.10.5
> 
> Do as follows:
> 
> `conda create -n libreface python=3.8` `conda activate libreface` `git clone https://github.com/ihp-lab/LibreFace.git` `cd LibreFace` `vi requirements_new.txt` Type in `:%s/0.9.3/0.10.5/g` then `:wq` `conda install ffmpeg` `pip install .`

why are there different requirements.txt?

**Comment by ac-alpha** [comment]
@6namdang `requirements_new.txt` (with the modifications that @joos2010kj suggested above) is what you need if you just plan to run inference through the Python package.

`requirements.txt` is what is required if you plan to train the models from scratch and might not be very up to date.

**Comment by ac-alpha** [comment]
Hello @Victor716 and others, we have a new release for libreface. It uses python3.9 and a more updated version of mediapipe. Please refer to updated README here for instructions to download -> https://github.com/ihp-lab/LibreFace#getting-started-with-python-installation

Let us know if the issue is resolved.

**Comment by ac-alpha** [comment]
Closing the issue. Please reopen if still facing this error.
